web/ui/react-app/src/pages/graph/MetricsExplorer.tsx

```diff
@@ -1,33 +1,48 @@
-import React, { Component } from 'react';
-import { Modal, ModalBody, ModalHeader } from 'reactstrap';
+import React, { Component, ChangeEvent } from 'react';
+import { Modal, ModalBody, ModalHeader, Input } from 'reactstrap';
 
 interface Props {
   show: boolean;
   updateShow(show: boolean): void;
   metrics: string[];
   insertAtCursor(value: string): void;
 }
-
-class MetricsExplorer extends Component<Props> {
+type MetricsExplorerState = {
+  searchTerm: string;
+};
+class MetricsExplorer extends Component<Props, MetricsExplorerState> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { searchTerm: '' };
+  }
+  handleSearchTerm = (event: ChangeEvent<HTMLInputElement>): void => {
+    this.setState({ searchTerm: event.target.value });
+  };
   handleMetricClick = (query: string): void => {
     this.props.insertAtCursor(query);
     this.props.updateShow(false);
+    this.setState({ searchTerm: '' });
   };
 
   toggle = (): void => {
     this.props.updateShow(!this.props.show);
   };
-
   render(): JSX.Element {
     return (
-      <Modal isOpen={this.props.show} toggle={this.toggle} className="metrics-explorer">
+      <Modal isOpen={this.props.show} toggle={this.toggle} className="metrics-explorer" scrollable>
         <ModalHeader toggle={this.toggle}>Metrics Explorer</ModalHeader>
         <ModalBody>
-          {this.props.metrics.map((metric) => (
-            <p key={metric} className="metric" onClick={this.handleMetricClick.bind(this, metric)}>
-              {metric}
-            </p>
-          ))}
+          <Input placeholder="Search" value={this.state.searchTerm} type="text" onChange={this.handleSearchTerm} />
+          {this.props.metrics
+            .filter((metric) => {
+              if (this.state.searchTerm === '') return true;
+              return metric.toLowerCase().includes(this.state.searchTerm.toLowerCase());
```

Comment:
Thanks! We already have search+filtering like this for flags on the /flags page (https://github.com/prometheus/prometheus/blob/143fce73cdc956b577e835a31e785244fc8a3551/web/ui/react-app/src/pages/flags/Flags.tsx), and I think we should keep the various filtering functionalities similar in the way they work. We use `@nexucis/fuzzy` for fuzzy searching + highlighting on the flags page, which I would recommend using here as well to keep things consistent.

Reply:
Took a crack at it, hope that's more like it!